### PR TITLE
feat(factory): return an `IUniversalRewardsDistributor`

### PR DESCRIPTION
**Pull request by julien-devatom**
Fixes: #45

### CODE REVIEW

File: src/UrdFactory.sol [hunk 1]
```diff
@@ -22,8 +23,8 @@ contract UrdFactory {
         bytes32 initialRoot,
         bytes32 initialIpfsHash,
         bytes32 salt
-    ) public returns (address urd) {
-        urd = address(
+    ) public returns (IUniversalRewardsDistributor urd) {
+        urd = IUniversalRewardsDistributor(
```
Comment (Rubilmax) [inline src/UrdFactory.sol:27]: I think typing it is unnecessary as `UniversalRewardsDistributor is IUniversalRewardsDistributor`

File: src/UrdFactory.sol [hunk 2]
```diff
@@ -2,6 +2,7 @@
 pragma solidity 0.8.21;
 
 import {UniversalRewardsDistributor} from "./UniversalRewardsDistributor.sol";
+import {IUniversalRewardsDistributor} from "./interfaces/IUniversalRewardsDistributor.sol";
```
Comment (Rubilmax) [inline src/UrdFactory.sol:5]: Then this is not used anymore ^^

**Review by Rubilmax — APPROVED** [review]

File: test/URDFactoryTest.sol [hunk 3]
```diff
@@ -27,10 +27,11 @@ contract UrdFactoryTest is Test {
...
             urdAddress, randomCaller, randomOwner, randomTimelock, randomRoot, randomIpfsHash, randomSalt
         );
-        address realAddress = factory.createUrd(randomOwner, randomTimelock, randomRoot, randomIpfsHash, randomSalt);
+        IUniversalRewardsDistributor urd =
+            factory.createUrd(randomOwner, randomTimelock, randomRoot, randomIpfsHash, randomSalt);
 
-        assertEq(realAddress, urdAddress);
-        IUniversalRewardsDistributor distributor = IUniversalRewardsDistributor(realAddress);
+        assertEq(address(urd), urdAddress);
+        IUniversalRewardsDistributor distributor = IUniversalRewardsDistributor(address(urd));
```
Comment (Rubilmax) [inline test/URDFactoryTest.sol:34]: This cast is no longer necessary

**Review by Rubilmax — CHANGES_REQUESTED** [review]

File: src/UrdFactory.sol (hunk 2 above)
Comment (julien-devatom) [reply src/UrdFactory.sol:5]: It's used in the return type but I can return a `UniversalRewardsDistributor` you're right

File: test/URDFactoryTest.sol (hunk 3 above)
Comment (julien-devatom) [reply test/URDFactoryTest.sol:34]: I was not aware of that lol

Comment (julien-devatom) [reply test/URDFactoryTest.sol:34]: False for L33, true for L34

Comment (Rubilmax) [reply test/URDFactoryTest.sol]: Yes, github automatically highlights the 3 previous lines...

**Review by Rubilmax — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**